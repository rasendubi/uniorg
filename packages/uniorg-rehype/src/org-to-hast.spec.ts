import { unified } from 'unified';
import orgParse from 'uniorg-parse';
import org2rehype from '.';
import raw from 'rehype-raw';
import format from 'rehype-format';
import html from 'rehype-stringify';
import type { OrgToHastOptions } from './org-to-hast';

const process = (input: string, options?: Partial<OrgToHastOptions>) => {
  const processor = unified()
    .use(orgParse)
    .use(org2rehype, options)
    .use(raw)
    .use(format)
    .use(html);

  return String(processor.processSync(input));
};

expect.addSnapshotSerializer({
  test(value) {
    return typeof value === 'string';
  },
  print(value) {
    return value as string;
  },
});

const hastTest = (
  name: string,
  input: string,
  options?: Partial<OrgToHastOptions>
) => {
  it(name, () => {
    const result = process(input, options);
    expect(result).toMatchSnapshot();
  });
};
hastTest.skip = (
  name: string,
  input: string,
  options?: Partial<OrgToHastOptions>
) => {
  it.skip(name, () => {
    const result = process(input, options);
    expect(result).toMatchSnapshot();
  });
};
hastTest.only = (
  name: string,
  input: string,
  options?: Partial<OrgToHastOptions>
) => {
  it.only(name, () => {
    const result = process(input, options);
    expect(result).toMatchSnapshot();
  });
};
hastTest.todo = (
  name: string,
  _input?: string,
  _options?: Partial<OrgToHastOptions>
) => {
  it.todo(name);
};

describe('org/org-to-hast', () => {
  hastTest('empty', ``);

  hastTest('paragraph', `hello`);

  hastTest('headline', `* hi`);

  hastTest(
    'multiple headlines',
    `* hi
** there
* hello
*** world
`
  );

  hastTest(
    'complex headline',
    `* TODO [#A] headline /italic/ title :some:tags:`
  );

  hastTest(
    'headline with sections',
    `hello
* hi
section
** hello
another section`
  );

  hastTest(
    'does not export :noexport: headline',
    `* hello
some text
* nope :noexport:
not exported text
`
  );

  hastTest('statistics cookie', `* hello [2/3]`);

  hastTest(
    'useSections',
    `hello
* headline 1
text
** headline 1.1
more text
* headline 2`,
    { useSections: true }
  );

  hastTest(
    'planning',
    `* headline
CLOSED: [2019-03-13 Wed 23:48] SCHEDULED: [2019-03-13 Wed] DEADLINE: [2019-03-14 Thu]`
  );

  hastTest(
    'property drawer',
    `* headline
:PROPERTIES:
:CREATED: [2019-03-13 Wed 23:57]
:END:`
  );

  hastTest(
    'custom drawer',
    `:MYDRAWER:
hello /there/
:END:`
  );

  hastTest('list', `- hello`);

  hastTest('ordered list', `1. one`);

  hastTest(
    'nested list',
    `
- hello
  - world
  - blah
- hi
`
  );

  hastTest(
    'description list',
    `- term1 :: description 1
- term 2 :: description 2`
  );

  // See https://github.com/rasendubi/uniorg/issues/15
  hastTest(
    'description list with complex tags',
    `
- [[https://example.com][Example]] :: Hello there!
- [[https://github.com][GitHub]] :: This is GitHub, your hub for Git repos.
- *Gitlab* :: Alternative to GitHub
- /Sourcehut/ :: Another alternative to GitHub that primarily uses email-based workflows.
- /Codeberg/ :: *ANOTHER ALTERNATIVE*
- /*self-hosting Git server*/ :: /*The ultimate Git solution for privacy-oriented individuals!*/
`
  );

  hastTest('link', `https://example.com`);

  hastTest('link mixed with text', `hello http://example.com blah`);

  hastTest(
    'escape local link',
    `[[./path with "spaces".org][path with "spaces"]]`
  );

  hastTest(
    'escape local link with percent',
    `[[file:local%2Bpath%2Bwith%2Bpercents.org][local path with percents]]`
  );

  hastTest(
    'do not double-escape urls',
    `[[http://localhost:3000/path%2Bwith%2Bpercents][path with percents]]`
  );

  hastTest(
    'src block',
    `#+begin_src c
,*a = b;
printf("%d\\n", *a);
#+end_src`
  );

  hastTest(
    'verse block',
    `#+BEGIN_VERSE
 Great clouds overhead
 Tiny black birds rise and fall
 Snow covers Emacs

    ---AlexSchroeder
#+END_VERSE`
  );

  hastTest(
    'center block',
    `#+begin_center
hello
#+end_center`
  );

  hastTest(
    'comment block',
    `#+begin_comment
hello
#+end_comment`
  );

  hastTest(
    'remove common src block offset',
    `
  #+begin_src
    hello
      world
  #+end_src`
  );

  hastTest(
    'example block',
    `#+begin_example
example
#+end_example`
  );

  hastTest(
    'export block',
    `#+begin_export
export
#+end_export`
  );

  hastTest(
    'export html block',
    `
#+begin_export html
<abbr title="World Health Organization">WHO</abbr> was founded in 1948.
#+end_export
`
  );

  hastTest(
    'export html keyword',
    `hello
#+HTML: <h1>html tag</h1>`
  );

  hastTest('export snippet', `@@html:<b>@@Hello, world!@@html:</b>@@`);

  hastTest(
    'special block',
    `#+begin_blah
hello
#+end_blah`
  );

  hastTest(
    'html5 fancy block',
    `#+begin_aside
hello
#+end_aside`
  );

  hastTest(
    'blockquote',
    `#+begin_quote
hello, world!
#+end_quote`
  );

  hastTest('keywords', `#+TITLE: blah`);

  hastTest(
    'emphasis',
    `/Consider/ ~t*h*e~ *following* =example= +strike+ _under_`
  );

  hastTest('superscript', `hello^there`);

  hastTest('subscript', `hello_there`);

  hastTest('images', `[[./image.png]]`);

  hastTest('does not export link as image', `[[./image.png][this is a link]]`);

  describe('#+ATTR_HTML', () => {
    // The results might be unexpected in this section, but that's how ox-html works.

    hastTest(
      'Simple paragraph -> attrs are attached to paragraph',
      `#+ATTR_HTML: :style color:red;
This is red paragraph.`
    );

    hastTest(
      'Simple image -> attrs are attached to image only',
      `#+ATTR_HTML: :alt Image alt
#+ATTR_HTML: :style width:320px;
#+ATTR_HTML: invalid :no-value :x y
[[./image.png]]`
    );

    hastTest(
      'Non-image link -> attrs are attached to both paragraph and link',
      `#+ATTR_HTML: :alt Link alt
[[./a-link.org]]`
    );

    hastTest(
      'Multiple links in a paragraph -> attrs are attached to paragraph and first link',
      `#+ATTR_HTML: :alt Link A alt
Some text [[./a-link.org]] [[./b-link.org]]`
    );

    hastTest(
      'On results block',
      `#+ATTR_HTML: :width 300px
#+RESULTS:
: sample output`
    );
  });

  describe('timestamps', () => {
    hastTest('inactive', `[2021-01-07 Thu]`);
    hastTest('inactive-range', `[2021-01-07 Thu]--[2021-01-08 Fri]`);
    hastTest('active', `<2021-01-07 Thu>`);
    hastTest('active-range', `<2021-01-07 Thu>--<2021-01-09 Sat>`);
    hastTest('with time', `[2021-01-07 Thu 19:36]`);
    hastTest('time range', `[2021-01-07 Thu 19:36-20:38]`);
    hastTest.todo('diary');
  });

  hastTest(
    'table',
    `
| head1  | head2 |
|--------+-------|
| value1 | value2 |
`
  );

  hastTest(
    'table without header',
    `
| value1 | value2 |
| value3 | value4 |
`
  );

  hastTest(
    'table, multiple body',
    `
| head1  | head2 |
|--------+-------|
| value1 | value2 |
| value3 | value4 |
|--------+--------|
| value5 | value6 |
`
  );

  hastTest(
    'table.el table',
    `
+------+
| blah |
+------+
`
  );

  hastTest(
    'comments',
    `hello
# comment
there`
  );

  hastTest('fixed-width', `: hello`);

  hastTest(
    'clock',
    `CLOCK: [2020-12-22 Tue 09:07]--[2020-12-22 Tue 11:10] =>  2:03`
  );

  hastTest(
    `latex environment`,
    `\\begin{hello}
some text
\\end{hello}`
  );

  hastTest('horizontal rule', `-----`);

  hastTest(
    'diary sexp',
    `%%(diary-anniversary 10 31 1948) Arthur's birthday (%d years old)`
  );

  describe('footnotes', () => {
    hastTest(
      'footnote-reference',
      `Some text with a footnote.[fn:1]
  
[fn:1] A very important footnote.`
    );

    hastTest(
      'footnote-definition',
      `Some text with a footnote.[fn:1]
Another footnote [fn:2]    
  
[fn:1] A very important footnote.
[fn:2] Another stellar footnote.`
    );

    hastTest('inline footnote', 'some text[fn:1: footnote definition]');

    hastTest(
      'maintains footnotes order',
      // Note that footnotes are emmitted in order of reference
      `footnote 2[fn:2]
footnote 1[fn:1]

[fn:1] second footnote
[fn:2] first footnote`
    );

    hastTest(
      'does not emit unreferenced footnotes',
      `footnote[fn:1]

[fn:1] hello
[fn:2] unreferenced`
    );

    hastTest('handles missing footnotes', 'footnote[fn:missing]');

    hastTest(
      'inline footnote',
      'footnotes[fn:label: inline footnote definition]'
    );

    hastTest(
      'anonymous inline footnote',
      'footnotes[fn:: inline footnote definition]'
    );
  });

  hastTest(
    'latex-fragment',
    `If $a^2=b$ and \\( b=2 \\), then the solution must be
either $$ a=+\\sqrt{2} $$ or \\[ a=-\\sqrt{2} \\].`
  );

  hastTest('entity', `\\Agrave`);

  hastTest(
    'citation',
    `[cite/style:common prefix; prefix @key suffix; @key2; common suffix]`
  );

  test('respects hProperties', () => {
    const s = unified()
      .use(orgParse)
      .use(() => (node: any) => {
        // org-data > section > headline
        const headline = node.children[0].children[0];
        headline.data = { hProperties: { id: 'my-custom-id' } };
      })
      // @ts-ignore for some reason the above does something weird
      .use(org2rehype)
      .use(format)
      .use(html)
      .processSync(`* headline`)
      .toString();

    expect(s).toMatchInlineSnapshot(`

      <h1 id="my-custom-id">headline</h1>

    `);
  });
});
