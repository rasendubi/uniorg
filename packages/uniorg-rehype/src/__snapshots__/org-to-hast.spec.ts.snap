// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`org/org-to-hast #+ATTR_HTML Multiple links in a paragraph -> attrs are attached to paragraph and first link 1`] = `

<p alt="Link A alt">Some text <a alt="Link A alt" href="./a-link.org">./a-link.org</a> <a href="./b-link.org">./b-link.org</a></p>

`;

exports[`org/org-to-hast #+ATTR_HTML Non-image link -> attrs are attached to both paragraph and link 1`] = `

<p alt="Link alt"><a alt="Link alt" href="./a-link.org">./a-link.org</a></p>

`;

exports[`org/org-to-hast #+ATTR_HTML On results block 1`] = `

<pre class="fixed-width" width="300px">sample output</pre>

`;

exports[`org/org-to-hast #+ATTR_HTML Simple image -> attrs are attached to image only 1`] = `

<p>
  <img alt="Image alt" style="width:320px;" x="y" src="./image.png">
</p>

`;

exports[`org/org-to-hast #+ATTR_HTML Simple paragraph -> attrs are attached to paragraph 1`] = `

<p style="color:red;">This is red paragraph.</p>

`;

exports[`org/org-to-hast blockquote 1`] = `

<blockquote>
  <p>hello, world!</p>
</blockquote>

`;

exports[`org/org-to-hast center block 1`] = `

<div class="center">
  <p>hello</p>
</div>

`;

exports[`org/org-to-hast citation 1`] = `

<p>common prefix prefix <a href="cite:key">cite:key</a> suffix<a href="cite:key2">cite:key2</a> common suffix</p>

`;

exports[`org/org-to-hast clock 1`] = ``;

exports[`org/org-to-hast comment block 1`] = ``;

exports[`org/org-to-hast comments 1`] = `

<p>hello</p>
<p>there</p>

`;

exports[`org/org-to-hast complex headline 1`] = `

<h1><span class="todo-keyword TODO">TODO</span> <span class="priority priority-A">[A]</span> headline <em>italic</em> title   <span class="tags"><span class="tag tag-some">some</span> <span class="tag tag-tags">tags</span></span></h1>

`;

exports[`org/org-to-hast custom drawer 1`] = ``;

exports[`org/org-to-hast description list 1`] = `

<dl>
  <dt>term1</dt>
  <dd>
    <p>description 1</p>
  </dd>
  <dt>term 2</dt>
  <dd>
    <p>description 2</p>
  </dd>
</dl>

`;

exports[`org/org-to-hast description list with complex tags 1`] = `

<dl>
  <dt><a href="https://example.com">Example</a></dt>
  <dd>
    <p>Hello there!</p>
  </dd>
  <dt><a href="https://github.com">GitHub</a></dt>
  <dd>
    <p>This is GitHub, your hub for Git repos.</p>
  </dd>
  <dt><strong>Gitlab</strong></dt>
  <dd>
    <p>Alternative to GitHub</p>
  </dd>
  <dt><em>Sourcehut</em></dt>
  <dd>
    <p>Another alternative to GitHub that primarily uses email-based workflows.</p>
  </dd>
  <dt><em>Codeberg</em></dt>
  <dd>
    <p><strong>ANOTHER ALTERNATIVE</strong></p>
  </dd>
  <dt><em><strong>self-hosting Git server</strong></em></dt>
  <dd>
    <p><em><strong>The ultimate Git solution for privacy-oriented individuals!</strong></em></p>
  </dd>
</dl>

`;

exports[`org/org-to-hast diary sexp 1`] = ``;

exports[`org/org-to-hast do not double-escape urls 1`] = `

<p><a href="http://localhost:3000/path%2Bwith%2Bpercents">path with percents</a></p>

`;

exports[`org/org-to-hast does not export :noexport: headline 1`] = `

<h1>hello</h1>
<p>some text</p>

`;

exports[`org/org-to-hast does not export link as image 1`] = `

<p><a href="./image.png">this is a link</a></p>

`;

exports[`org/org-to-hast emphasis 1`] = `

<p><em>Consider</em> <code class="inline-code">t*h*e</code> <strong>following</strong> <code class="inline-verbatim">example</code> <del>strike</del> <span class="underline" style="text-decoration: underline;">under</span></p>

`;

exports[`org/org-to-hast empty 1`] = ``;

exports[`org/org-to-hast entity 1`] = `

<p>À</p>

`;

exports[`org/org-to-hast escape local link 1`] = `

<p><a href="./path%20with%20%22spaces%22.org">path with "spaces"</a></p>

`;

exports[`org/org-to-hast escape local link with percent 1`] = `

<p><a href="file:local%252Bpath%252Bwith%252Bpercents.org">local path with percents</a></p>

`;

exports[`org/org-to-hast example block 1`] = `

<div class="example">example</div>

`;

exports[`org/org-to-hast export block 1`] = ``;

exports[`org/org-to-hast export html block 1`] = `<abbr title="World Health Organization">WHO</abbr> was founded in 1948.`;

exports[`org/org-to-hast export html keyword 1`] = `

<p>hello</p>
<h1>html tag</h1>

`;

exports[`org/org-to-hast export snippet 1`] = `

<p><b>Hello, world!</b></p>

`;

exports[`org/org-to-hast fixed-width 1`] = `

<pre class="fixed-width">hello</pre>

`;

exports[`org/org-to-hast footnotes anonymous inline footnote 1`] = `

<p>footnotes<sup><a href="#fn.1" class="footref" id="fnr.1" role="doc-backlink">1</a></sup></p>
<h1>Footnotes:</h1>
<div class="footnote-definition"><sup><a class="footnum" id="fn.1" href="#fnr.1" role="doc-backlink">1</a></sup>
  <div class="footdef" role="doc-footnote">inline footnote definition</div>
</div>

`;

exports[`org/org-to-hast footnotes does not emit unreferenced footnotes 1`] = `

<p>footnote<sup><a href="#fn.1" class="footref" id="fnr.1" role="doc-backlink">1</a></sup></p>
<h1>Footnotes:</h1>
<div class="footnote-definition"><sup><a class="footnum" id="fn.1" href="#fnr.1" role="doc-backlink">1</a></sup>
  <div class="footdef" role="doc-footnote">
    <p>hello</p>
  </div>
</div>

`;

exports[`org/org-to-hast footnotes footnote-definition 1`] = `

<p>
  Some text with a footnote.<sup><a href="#fn.1" class="footref" id="fnr.1" role="doc-backlink">1</a></sup>
  Another footnote <sup><a href="#fn.2" class="footref" id="fnr.2" role="doc-backlink">2</a></sup>
</p>
<h1>Footnotes:</h1>
<div class="footnote-definition"><sup><a class="footnum" id="fn.1" href="#fnr.1" role="doc-backlink">1</a></sup>
  <div class="footdef" role="doc-footnote">
    <p>A very important footnote.</p>
  </div>
</div>
<div class="footnote-definition"><sup><a class="footnum" id="fn.2" href="#fnr.2" role="doc-backlink">2</a></sup>
  <div class="footdef" role="doc-footnote">
    <p>Another stellar footnote.</p>
  </div>
</div>

`;

exports[`org/org-to-hast footnotes footnote-reference 1`] = `

<p>Some text with a footnote.<sup><a href="#fn.1" class="footref" id="fnr.1" role="doc-backlink">1</a></sup></p>
<h1>Footnotes:</h1>
<div class="footnote-definition"><sup><a class="footnum" id="fn.1" href="#fnr.1" role="doc-backlink">1</a></sup>
  <div class="footdef" role="doc-footnote">
    <p>A very important footnote.</p>
  </div>
</div>

`;

exports[`org/org-to-hast footnotes handles missing footnotes 1`] = `

<p>footnote<sup><a href="#fn.1" class="footref" id="fnr.1" role="doc-backlink">1</a></sup></p>

`;

exports[`org/org-to-hast footnotes inline footnote 1`] = `

<p>some text<sup><a href="#fn.1" class="footref" id="fnr.1" role="doc-backlink">1</a></sup></p>
<h1>Footnotes:</h1>
<div class="footnote-definition"><sup><a class="footnum" id="fn.1" href="#fnr.1" role="doc-backlink">1</a></sup>
  <div class="footdef" role="doc-footnote">footnote definition</div>
</div>

`;

exports[`org/org-to-hast footnotes inline footnote 2`] = `

<p>footnotes<sup><a href="#fn.1" class="footref" id="fnr.1" role="doc-backlink">1</a></sup></p>
<h1>Footnotes:</h1>
<div class="footnote-definition"><sup><a class="footnum" id="fn.1" href="#fnr.1" role="doc-backlink">1</a></sup>
  <div class="footdef" role="doc-footnote">inline footnote definition</div>
</div>

`;

exports[`org/org-to-hast footnotes maintains footnotes order 1`] = `

<p>
  footnote 2<sup><a href="#fn.1" class="footref" id="fnr.1" role="doc-backlink">1</a></sup>
  footnote 1<sup><a href="#fn.2" class="footref" id="fnr.2" role="doc-backlink">2</a></sup>
</p>
<h1>Footnotes:</h1>
<div class="footnote-definition"><sup><a class="footnum" id="fn.1" href="#fnr.1" role="doc-backlink">1</a></sup>
  <div class="footdef" role="doc-footnote">
    <p>first footnote</p>
  </div>
</div>
<div class="footnote-definition"><sup><a class="footnum" id="fn.2" href="#fnr.2" role="doc-backlink">2</a></sup>
  <div class="footdef" role="doc-footnote">
    <p>second footnote</p>
  </div>
</div>

`;

exports[`org/org-to-hast headline 1`] = `

<h1>hi</h1>

`;

exports[`org/org-to-hast headline with sections 1`] = `

<p>hello</p>
<h1>hi</h1>
<p>section</p>
<h2>hello</h2>
<p>another section</p>

`;

exports[`org/org-to-hast horizontal rule 1`] = `

<hr>

`;

exports[`org/org-to-hast html5 fancy block 1`] = `

<aside>
  <p>hello</p>
</aside>

`;

exports[`org/org-to-hast images 1`] = `

<p>
  <img src="./image.png">
</p>

`;

exports[`org/org-to-hast keywords 1`] = ``;

exports[`org/org-to-hast latex environment 1`] = `

<div class="math math-display">
  \\begin{hello}
  some text
  \\end{hello}
</div>

`;

exports[`org/org-to-hast latex-fragment 1`] = `

<p>
  If <span class="math math-inline">a^2=b</span> and <span class="math math-inline">b=2</span>, then the solution must be
  either <span class="math math-inline">a=+\\sqrt{2}</span> or <span class="math math-inline">a=-\\sqrt{2}</span>.
</p>

`;

exports[`org/org-to-hast line-break 1`] = `

<p>hello<br>world</p>

`;

exports[`org/org-to-hast link 1`] = `

<p><a href="https://example.com">https://example.com</a></p>

`;

exports[`org/org-to-hast link mixed with text 1`] = `

<p>hello <a href="http://example.com">http://example.com</a> blah</p>

`;

exports[`org/org-to-hast list 1`] = `

<ul>
  <li>
    <p>hello</p>
  </li>
</ul>

`;

exports[`org/org-to-hast multiple headlines 1`] = `

<h1>hi</h1>
<h2>there</h2>
<h1>hello</h1>
<h3>world</h3>

`;

exports[`org/org-to-hast nested list 1`] = `

<ul>
  <li>
    <p>hello</p>
    <ul>
      <li>
        <p>world</p>
      </li>
      <li>
        <p>blah</p>
      </li>
    </ul>
  </li>
  <li>
    <p>hi</p>
  </li>
</ul>

`;

exports[`org/org-to-hast ordered list 1`] = `

<ol>
  <li>
    <p>one</p>
  </li>
</ol>

`;

exports[`org/org-to-hast paragraph 1`] = `

<p>hello</p>

`;

exports[`org/org-to-hast planning 1`] = `

<h1>headline</h1>

`;

exports[`org/org-to-hast property drawer 1`] = `

<h1>headline</h1>

`;

exports[`org/org-to-hast remove common src block offset 1`] = `

<pre class="src-block"><code>hello
  world
</code></pre>

`;

exports[`org/org-to-hast special block 1`] = `

<div class="special-block block-blah">
  <p>hello</p>
</div>

`;

exports[`org/org-to-hast src block 1`] = `

<pre class="src-block"><code class="language-c">*a = b;
printf("%d\\n", *a);
</code></pre>

`;

exports[`org/org-to-hast statistics cookie 1`] = `

<h1>hello <span class="statistics-cookie">[2/3]</span></h1>

`;

exports[`org/org-to-hast subscript 1`] = `

<p>hello<sub>there</sub></p>

`;

exports[`org/org-to-hast superscript 1`] = `

<p>hello<sup>there</sup></p>

`;

exports[`org/org-to-hast table 1`] = `

<table>
  <thead>
    <tr>
      <th>head1</th>
      <th>head2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>value1</td>
      <td>value2</td>
    </tr>
  </tbody>
</table>

`;

exports[`org/org-to-hast table without header 1`] = `

<table>
  <tbody>
    <tr>
      <td>value1</td>
      <td>value2</td>
    </tr>
    <tr>
      <td>value3</td>
      <td>value4</td>
    </tr>
  </tbody>
</table>

`;

exports[`org/org-to-hast table, multiple body 1`] = `

<table>
  <thead>
    <tr>
      <th>head1</th>
      <th>head2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>value1</td>
      <td>value2</td>
    </tr>
    <tr>
      <td>value3</td>
      <td>value4</td>
    </tr>
  </tbody>
  <tbody>
    <tr>
      <td>value5</td>
      <td>value6</td>
    </tr>
  </tbody>
</table>

`;

exports[`org/org-to-hast table.el table 1`] = `

<pre class="table-el">+------+
| blah |
+------+
</pre>

`;

exports[`org/org-to-hast timestamps active 1`] = `

<p><span class="timestamp">&#x3C;2021-01-07 Thu></span></p>

`;

exports[`org/org-to-hast timestamps active-range 1`] = `

<p><span class="timestamp">&#x3C;2021-01-07 Thu>--&#x3C;2021-01-09 Sat></span></p>

`;

exports[`org/org-to-hast timestamps inactive 1`] = `

<p><span class="timestamp">[2021-01-07 Thu]</span></p>

`;

exports[`org/org-to-hast timestamps inactive-range 1`] = `

<p><span class="timestamp">[2021-01-07 Thu]--[2021-01-08 Fri]</span></p>

`;

exports[`org/org-to-hast timestamps time range 1`] = `

<p><span class="timestamp">[2021-01-07 Thu 19:36-20:38]</span></p>

`;

exports[`org/org-to-hast timestamps with time 1`] = `

<p><span class="timestamp">[2021-01-07 Thu 19:36]</span></p>

`;

exports[`org/org-to-hast useSections 1`] = `

<p>hello</p>
<section class="section-level-1">
  <h1>headline 1</h1>
  <p>text</p>
  <section class="section-level-2">
    <h2>headline 1.1</h2>
    <p>more text</p>
  </section>
</section>
<section class="section-level-1">
  <h1>headline 2</h1>
</section>

`;

exports[`org/org-to-hast verse block 1`] = `

<pre class="verse"> Great clouds overhead
 Tiny black birds rise and fall
 Snow covers Emacs

    ---AlexSchroeder
</pre>

`;
